proxy/src/main/java/com/wavefront/agent/ChannelByteArrayHandler.java

```diff
@@ -63,9 +56,17 @@ protected void channelRead0(ChannelHandlerContext ctx, byte[] msg) throws Except
     try {
       decoder.decodeReportPoints(msg, points, "dummy");
       for (final ReportPoint point: points) {
-        if (!this.metricPredicate.apply(point.getMetric())) {
-          pointHandler.handleBlockedPoint(point.getMetric());
-          continue;
+        if (preprocessor != null) {
+          // backwards compatibility: apply "pointLine" rules to metric name
+          if (!preprocessor.forPointLine().filter(point.getMetric())) {
+            pointHandler.handleBlockedPoint(preprocessor.forReportPoint().getLastFilterResult());
+            continue;
+          }
+          preprocessor.forReportPoint().transform(point);
```

Comment:
it looks like this block of code is the entry point to all stream processing you've added in this PR, is that correct?